### ISSUE TITLE
Preact Support?

### ISSUE THREAD

**Issue by Chrischuck** [AliasIO/Wappalyzer#1502]
Would it be possible to add the [Preact](https://github.com/developit/preact) library? When using React and then aliasing it to Preact using  [Preact-compat](https://github.com/developit/preact-compat) to convert it, the react icon stops showing up.

Thanks!

**Comment by jvoisin** [comment]
Be our guest and do a pull-request, it's not that hard :)

**Comment by amilajack** [comment]
+!

**Comment by wardoost** [comment]
I don't think it will be easy to detect this one. I'll compare with React which can be detected by

- **script pattern**: As Preact is so small it's very likely it's bundled when used. Otherwise you're kind of defying the purpose of Preact no?
- **React global**: When you bundle with Webpack you don't have those globals anymore.
- **`data-react` attribute in html**: Preact doesn't use an attribute like this for it's root element.

Correct me if I'm wrong here as I have only briefly tested Preact. Btw, you're not using React anymore if your aliasing it to Preact-compat. So React should not show up on an app that uses preact/preact-compat.

**Comment by amilajack** [comment]
@developit

**Comment by developit** [comment]
```js
function isPreact() {
  var all = document.querySelectorAll('*');
  for (let i=all.length; i--; ) {
    if (typeof Symbol!=='undefined' && all[i][Symbol.for('preactattr')] || all[i].__preactattr_) {
      return true;
    }
  }
  return false;
}
```

You can test it out on preactjs.com or webpack.js.org.

**Comment by AliasIO** [comment]
You can't run arbitrary JavaScript on webpages with Wappalyzer.

**Comment by wardoost** [comment]
@AliasIO Do you think this would be too heavy to run if a lot of entries would do this or is it technically not feasible for another reason?

**Comment by developit** [comment]
do you guys scan source files? if so `__preactattr_` is a pretty good key.

**Comment by wardoost** [comment]
According to the [specification](https://github.com/AliasIO/Wappalyzer/wiki/Specification) not. I think we should look for a way to scan JS and source files, especially with the popularity of bundlers. The extensions would definitely need to use a web worker then as requested in #362.

**Comment by AliasIO** [comment]
Wappalyzer only deals with the main page request, I wouldn't want to run the regular expressions over every file.

**Comment by wardoost** [comment]
I think this issue can be closed then as there is no alternative way to detect Preact. Or would you leave it open in case someone finds another solution in the future?
